post-processing/test_post_processing.py

```diff
@@ -0,0 +1,132 @@
+import os
+from pathlib import Path
+import post_processing as post
+import re
+import shutil
+import subprocess as sp
+
+# Run given benchmark with reframe using subprocess
+def run_benchmark(path):
+    sp.run("reframe -c " + path + " -r", shell=True)
+
+# Clean unnecessary files and folders at end of test after running reframe
+def benchmark_cleanup(benchmark_run, remove_test_logs):
+
+    # remove residual folders if test benchmark has been run
+    if benchmark_run:
+        shutil.rmtree("stage", ignore_errors=True)
+        shutil.rmtree("output", ignore_errors=True)
+
+    # remove test log files
+    if remove_test_logs:
+        shutil.rmtree("perflogs", ignore_errors=True)
+
+# Remove given field from perflog line
+def remove_field_from_perflog(line, field_index):
+
+    # parse line
+    columns = line.strip().split("|")
+    # remove given field
+    columns.pop(field_index)
+    # convert back to string
+    line = "|".join(columns)
+
+    return line
+
+def test_display_name_parsing():
+
+    display_name = "TestName %param1=one %param2=two %param3=three"
+    params = post.get_params_from_name(display_name)
+
+    # check param length, names, and values
+    assert len(params) == 3
+    assert (params["param1"] == "one") & (params["param2"] == "two") & (params["param3"] == "three")
+
+def test_read_perflog():
+
+    REQUIRED_FIELD = "flops_value"
+    EXPECTED_FIELDS = ["job_completion_time", "version", "info", "jobid", "num_tasks", "num_cpus_per_task", "num_tasks_per_node", "num_gpus_per_node", "flops_value", "flops_unit", "flops_ref", "flops_lower_thres", "flops_upper_thres", "spack_spec", "display_name", "system", "partition", "environ", "extra_resources", "env_vars", "tags"]
+
+    # set True to remove test log files at the end of the test
+    remove_test_logs = True
+    # keeps track of whether the test benchmark has been run during this test
+    benchmark_run = False
+
+    # absolute path to post-processing directory
+    post_processing_dir = Path(__file__).parent
+    # change current working dir to avoid
+    # potentially deleting non-test perflogs later
+    os.chdir(post_processing_dir)
+    # resolve relative path from post-processing directory to access sombrero benchmark
+    sombrero_bench_path = str((post_processing_dir / "../benchmarks/examples/sombrero/sombrero.py").resolve())
+
+    # get test log file paths
+    perflog_path = os.path.join(os.getcwd(),"perflogs/default/default/")
+    sombrero_incomplete_log = "SombreroBenchmarkIncomplete.log"
+    sombrero_incomplete_log_path = os.path.join(perflog_path, sombrero_incomplete_log)
+
+    # look for existing log files
+    sombrero_logs = []
+    sombrero_log_re = re.compile(r"^SombreroBenchmark\_\w{8}\.log$")
+    for subdirs, dirs, files in os.walk(perflog_path):
+        sombrero_logs = list(filter(sombrero_log_re.match, files))
+
+    try:
+        # to save time don't re-generate test log files if they already exist
+        if not sombrero_logs:

```

Comment:
Remove and regenerate logs in the beginning of the test every time.
The optional deletion at the end of the test is fine as is.